sklearn/feature_selection/_univariate_selection.py

```diff
@@ -355,8 +361,16 @@ def f_regression(X, y, *, center=True):
     deg_of_freedom = y.size - (2 if center else 1)
 
     corr_coef_squared = correlation_coefficient ** 2
-    f_statistic = corr_coef_squared / (1 - corr_coef_squared) * deg_of_freedom
-    p_values = stats.f.sf(f_statistic, 1, deg_of_freedom)
+
+    with np.errstate(divide="ignore", invalid="ignore"):
+        f_statistic = corr_coef_squared / (1 - corr_coef_squared) * deg_of_freedom
+        p_values = stats.f.sf(f_statistic, 1, deg_of_freedom)
+
+    # The Pearson correlation coefficient is undefined for random variables
+    # (attributes and target) with a standard deviation of zero
+    nan_mask = np.isnan(f_statistic)
+    f_statistic[nan_mask] = 0.0
+    p_values[nan_mask] = 1.0
```

Comment:
Technically this a change in behavior and we should document that we are doing this type of masking.